src/WebAPI/Routes/ConversationHistoryRoutes.cs

```diff
@@ -40,24 +36,20 @@ public static void MapConversationRoutes(this WebApplication app)
         routeGroup
             .MapPost(
                 "/Conversation",
-                async (HttpContext context, string conversation, string firstMessage) =>
+                async (ChatHistoryService historyService, string conversation, string firstMessage) =>
                 {
-                    //TODO: Return ID of newly created row to frontend
-                    
-                    var service = context.RequestServices.GetRequiredService<ChatHistoryService>();
-                    await service.AddConversation(conversation, firstMessage);
+                    var id = await historyService.AddConversation(conversation, firstMessage);
+                    return TypedResults.Ok(id);
                 })
             .WithName("AddConversationHistory")
             .RequireAuthorization(ChatHistoryPolicy);
         
         routeGroup
             .MapPut(
-                "/Conversation", 
-                async (HttpContext context, int id, string conversation) =>
+                "/Conversation{id}", 
```

Comment:
Update the route to follow the standard format:

```suggestion
                "/Conversation/{id}", 
```